pddl/logic/effects.py

```diff
@@ -12,58 +12,17 @@
 
 """This modules implements PDDL effects."""
 import functools
-from typing import AbstractSet, Collection, Generic, Optional, Sequence, TypeVar, Union
+from typing import AbstractSet, Collection, Optional, TypeVar, Union
 
 from pddl.helpers.base import _typed_parameters, ensure_set
 from pddl.helpers.cache_hash import cache_hash
 from pddl.logic import Variable
-from pddl.logic.base import Atomic, Formula, Not, OneOf
+from pddl.logic.base import And, Atomic, Formula, Not, OneOf
 from pddl.parser.symbols import Symbols
 
 EffectType = TypeVar("EffectType")
 
 
-@cache_hash
-@functools.total_ordering
```

Comment:
We lose `functools.total_ordering` in this shift. That ok?

Reply:
This should be ok as we get `__eq__` from the `BinaryOp,` but it might be a problem if we need an ordering of operands like `__lt__.`

Reply:
Ok. Until we need it, fine without it ;).